app/components/profile.jsx

```diff
@@ -94,13 +85,16 @@ export class Profile extends Component {
 
 }
 
-function mapStateToProps ({ session }) {
-  return { session };
+function mapStateToProps ({ users }, { params: { username } }) {
```

Comment:
see the comma we use here? thats splitting up the two arguments that're being passed in (see docs [here](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments), though you dont have to). the first argument is the global state, the second is the react component's original props (the ones that're being passed to it). this part is a bit tricky, but we actually want to get the current user using `currentUser`, not `username`. the reason for this is because we want `user` to change as soon as possible, so that begs the question, which one updates first, `currentUser` or `username`. well `currentUser` changes when we call `setCurrentUser` which is in the `reset` function and `username` will update when the routing navigation fully completes, which is after `componentWillReceiveProps` (and `reset`) is finished running. so that means `currentUser` changes first. thats why we should use that. and since `currentUser` is in the global state, we don't even need the second arg anymore.

final result:

```js
function mapStateToProps ({ currentUser, users }) {
```

you should also do this in `tracker.jsx` (line 84)